### ISSUE TITLE
Allow passing --user to adb when starting app (launching apps in work profile)

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#20324]
Internal: b/142944798

`flutter run` cannot pass `--user` flag to adb launcher.  

https://github.com/flutter/flutter/blob/8dd06a119015b1053df0b9af564e14684f7ca995/packages/flutter_tools/lib/src/android/android_device.dart#L414-L420

That means users can't select to launch their apps in work profile vs personal profile. This is critical for any enterprise app that needs to be tested in this mode.

See work profile section in https://developer.android.com/studio/run/.

**Comment by marcjoha** [comment]
Any resolution or work-around in sight? Flutter is deploying into my device's work profile although I want the app to land in my personal user profile.

**Comment by AlbertoMonteiro** [comment]
@marcjoha I had same problem, I changed the flutter_tools code and [deleted the flutter_tools snapshot](https://github.com/flutter/flutter/wiki/The-flutter-tool#making-changes-to-the-flutter-tool)

I ran this command
```
adb.exe shell pm list users
```
Then I got that my Work profile is user id 10

Then runned `flutter` in terminal, so it rebuilded the flutter_tools that does have the code tu run adb

File changed: **packages/flutter_tools/lib/src/android/android_device.dart**
![image](https://user-images.githubusercontent.com/836496/78295391-73b75300-7502-11ea-9f6f-27d64fbb8592.png)

**Comment by jmagman** [comment]
`flutter run`, `install`, `attach`, and `drive` now have a `--device-user` flag.

**Comment by nbsathawara** [comment]
Hi,

I've two profile in my phone one is personal and one is work.

Here is the list of users which I get by running "adb shell pm list users"
UserInfo{0:Nikhil:13} running
UserInfo{10:Work profile:30} running

Now I want to test my flutter app into work profile.

Generally as per documentation any app which run into multiuser device should install in all profiles.
But for me its only installing into my personal profile not into work profile.

I've tried suggested changes and run following command;
"flutter run --device-user 10"

But it gives me the following error;
"Shell does not have permission to access user 10"

I've also created new issue #81115.

Any help is appreciated.
Thanks in advance.
